### PR TITLE
Moves incident edit to modal and adds a dynamic tag menu

### PR DESCRIPTION
**Pull request by kevgliss**

### CODE REVIEW

File: src/dispatch/plugins/dispatch_slack/modals.py [hunk 1]
```diff
@@ -485,6 +589,228 @@ def update_update_participant_modal(
...
+    modal = {
+        "type": "modal",
+        "title": {"type": "plain_text", "text": "Edit Incident"},
+        "blocks": [
+            {
+                "type": "context",
+                "elements": [
+                    {
+                        "type": "mrkdwn",
+                        "text": "Use this form to edit the details of incident.",
```
Comment (mvilanova) [inline src/dispatch/plugins/dispatch_slack/modals.py:660]:
```suggestion
                        "text": "Use this form to update the details of the incident.",
```

File: src/dispatch/plugins/dispatch_slack/modals.py [hunk 2]
```diff
@@ -485,6 +589,228 @@ def update_update_participant_modal(
...
+                "text": {"type": "plain_text", "text": f"{t.tag_type.name}/{t.name}"},
+                "value": str(
+                    t.id
+                ),  # NOTE slack doesn't not accept int's as values (fails silently)
+            }
+        )
+
+    modal = {
+        "type": "modal",
+        "title": {"type": "plain_text", "text": "Edit Incident"},
```
Comment (mvilanova) [inline src/dispatch/plugins/dispatch_slack/modals.py:653]:
```suggestion
        "title": {"type": "plain_text", "text": "Update Incident"},
```

File: src/dispatch/plugins/dispatch_slack/modals.py [hunk 3]
```diff
@@ -485,6 +589,228 @@ def update_update_participant_modal(
...
+                    "initial_options": selected_tags,
+                    "action_id": Menus.tags,
+                    "type": "multi_external_select",
+                    "placeholder": {"type": "plain_text", "text": "Select related tags"},
+                    "min_query_length": 3,
+                },
+            },
+        ],
+        "close": {"type": "plain_text", "text": "Cancel"},
+        "submit": {"type": "plain_text", "text": "Submit"},
```
Comment (mvilanova) [inline src/dispatch/plugins/dispatch_slack/modals.py:761]:
```suggestion
        "submit": {"type": "plain_text", "text": "Update"},
```

File: src/dispatch/plugins/dispatch_slack/modals.py [hunk 4]
```diff
@@ -485,6 +589,228 @@ def update_update_participant_modal(
...
+    channel_id: str,
+    incident_id: int,
+    action: dict,
+    db_session=None,
+    slack_client=None,
+):
+    """Massages slack dialog data into something that Dispatch can use."""
+    submitted_form = action.get("view")
+    parsed_form_data = parse_submitted_form(submitted_form)
+    print(parsed_form_data)
```
Comment (mvilanova) [inline src/dispatch/plugins/dispatch_slack/modals.py:789]: Remove

**Review by mvilanova — APPROVED** [review]
Looks good. Just a few suggestions. Is it rendering the selectors on the right hand side now due to the size of the tags field? I liked it better when the selection was under its section title.

**Comment by kevgliss** [comment]
> Looks good. Just a few suggestions. Is it rendering the selectors on the right hand side now due to the size of the tags field? I liked it better when the selection was under its section title.

No idea, I think it just might be a change because of modals? I didn't specify any layout stuff.

**Pull request merged**